src/components/pickers/BasePicker.tsx

```diff
@@ -178,8 +182,22 @@ export class BasePicker<T, P extends IBasePickerProps<T>> extends BaseComponent<
       if (Array.isArray(suggestionsArray)) {
         this.resolveNewValue(updatedValue, suggestionsArray);
       } else if (suggestionsPromiseLike.then) {
+        if (!this.loadingTimer) {
```

Comment:
I'm a little nervous about the logic here, maybe we could just sync quickly on it; when the user types, are we debouncing input, or are we just debouncing the loading state? It seems like the later; the BasePicker owns the suggestionsLoading boolean and delays sets it to true while the promise is executing.